python/tank/deploy/descriptor.py

```diff
@@ -436,35 +436,43 @@ class VersionedSingletonDescriptor(AppDescriptor):
     """
     _instances = dict()
 
-    def __new__(cls, pc_path, bundle_install_path, location_dict, *args, **kwargs):
-        # We will cache based on the bundle install path, name of the
-        # app/engine/framework, and version number.
+    def __new__(cls, pc_path, bundle_cache_root, location_dict, *args, **kwargs):
+        """
+        Executed prior to all __init__s. Handles singleton caching of descriptors.
```

Comment:
Very picky, but I would say "Executed prior to all __init__ calls." to make it look prettier. ;)